### ISSUE TITLE
Suggestion: allow arbitrary additional opt in field and display it in doc

### ISSUE THREAD

**Issue by tanguilp** [Qqwy/elixir_confy#4]
The field macro could allow any option and display it after the default value.

This would look like:

```elixir
field :some_interval, :nonnegative_integer, default: 30, unit: "seconds"

field :scope_configuration, :term, type: "t:SomeModule.some_type/0", configured_at: "compilation time"
```
Ideally:
- the atom key would be converted to text ("Unit:", "Type:", "Configured at:")
- the value would be a "interpretated" string, so that one could insert type definitions (such as in the example above), `c:Module.callback` and so on
- these fields would be displayed in alphabetical order to retain consistency, or even in a separate "Additional properties" section

What do you think?

**Comment by Qqwy** [comment]
I do not think this is a good idea, because if I understand your proposal correctly, the 'additional options' are very much free-form, without making it clear what additional options are (dis)allowed.

Instead I think it would be better to add these additional options as extra fields to the specify definition, or possibly change the types to more sophisticated ones (like e.g. having an `%Interval{}` struct).

Does that make sense?

Related to this, I would like to introduce a possibility of nested specifications. I am however not sure yet how this should look/behave exactly.

**Comment by tanguilp** [comment]
That's actually just free-form fields, and the main idea is to have them displayed in a table which would be cleaner IMO than the actual way of displaying it. But I agree it's not that important :)
